### ISSUE TITLE
Error saving app state:  TypeError: immutableWndPayload.get is not a function

### ISSUE THREAD

**Issue by luixxiul** [brave/browser-laptop#10485]
**Describe the issue you encountered:** 

````
Error saving app state:  TypeError: immutableWndPayload.get is not a function
    at immutablePayload.get.forEach (/usr/lib/brave/resources/app.asar/app/sessionStore.js:83:44)
    at List.__iterate (/usr/lib/brave/resources/app.asar/node_modules/immutable/dist/immutable.js:2208:13)
    at List.forEach (/usr/lib/brave/resources/app.asar/node_modules/immutable/dist/immutable.js:4383:19)
    at Promise (/usr/lib/brave/resources/app.asar/app/sessionStore.js:82:46)
    at Promise (<anonymous>)
    at Object.module.exports.saveAppState (/usr/lib/brave/resources/app.asar/app/sessionStore.js:76:10)
    at saveAppState (/usr/lib/brave/resources/app.asar/app/sessionStoreShutdown.js:88:23)
    at sessionStoreQueue.push (/usr/lib/brave/resources/app.asar/app/sessionStoreShutdown.js:140:7)
    at async.queue (/usr/lib/brave/resources/app.asar/app/sessionStoreShutdown.js:54:5)
    at /usr/lib/brave/resources/app.asar/node_modules/async/dist/async.js:4069:9
````

Brave: 0.18.22 
rev: d850e287ace296e2c62e28ba2979cf172e41cc45 
Muon: 4.3.9 
libchromiumcontent: 60.0.3112.90 
V8: 6.0.286.52 
Node.js: 7.9.0 
Update Channel: dev 
OS Platform: Linux 
OS Release: 4.9.0-3-amd64 
OS Architecture: x64

- Steps to reproduce:
    1. Open Terminal
    2. `brave`
    3. Close the browser
    
- Actual result: See above

- Expected result: That error should not happen

- Extra QA steps:
    1.
    2.
    3.

- Any related issues: https://github.com/brave/browser-laptop/issues/10425

**Comment by bbondy** [comment]
Are there steps from a fresh profile? I wonder if it is from an old bug that was fixed and can't happen anymore.  I'm setting up a linux vm to try now in the meantime.

**Comment by luixxiul** [comment]
For me the issue can be reproduced with a fresh profile.

**Comment by bbondy** [comment]
Nice find @luixxiul! 

While testing on Linux I seen an unrelated error with deleting extensions. I think it was the cause to lead to this.   

I could only reproduce with packaged builds and I confirmed I could reproduce with a package from my dev machine by packaging and testing.  
Then I changed to the version with that fix and it did not reproduce after when doing a packaged build.


Fix here: 0a599c6f1d7f05ad94a92fc27a67be20047b9a83

Please retest with 0.18.23 or later.
